chemprop/uncertainty/predictor.py

```diff
@@ -46,11 +49,25 @@ def _calc_prediction_uncertainty(
 
 @UncertaintyPredictorRegistry.register("ensemble")
 class EnsemblePredictor(UncertaintyPredictor):
-    def _calc_prediction_uncertainty(
-        self, dataloader: DataLoader, models: Iterable[MPNN], trainer: pl.Trainer
-    ) -> tuple[Tensor, Tensor]:
-        ...
-        return
+    """
+    Class that predicts uncertainty for predictions based on the variance in predictions among
+    an ensemble's submodels.
+    """
+    def _calc_prediction_uncertainty(self, dataloader, models, trainer) -> Tensor:
+        if len(models) <= 1:
+            raise ValueError(
+                "Ensemble method for uncertainty is only available when multiple models are provided."
+            )
+        individual_preds = []
+        for model in models:
+            predss = trainer.predict(model, dataloader)
+            preds = torch.concat(predss, 0)
+            if isinstance(model.predictor, MulticlassClassificationFFN):
+                preds = torch.argmax(preds, dim=-1)
+            individual_preds.append(preds)
+        stacked_preds = torch.stack(individual_preds).float()
+        vars = torch.var(stacked_preds, dim=0, correction=0)
+        return stacked_preds, vars
```

Comment:
```suggestion
        ensemble_preds = torch.stack(ensemble_preds).float()
        vars = ensemble_preds.var(dim=0, correction=0)
        
        return stacked_preds, vars
```
Is there a reason to return the raw predictions and the uncertainty metric? I can't tell from the class definition what contract this method signs.

Reply:
As in the predict CLI, we save the prediction results for each model. To make this still be available in the CLI after  implementing uncertainty methods, we opt to return individual results, which can then be used to calculate mean values within the CLI module. The output is thus ``(n, t, m)`` for regression and ``(n, t, c, m)`` for multiclass classification, where ``n`` is the number of inputs, ``t`` is the number of tasks, ``c`` is the number of classes, and ``m`` is the number of models. I will update the docstring in the #986 PR.